sympy/matrices/tests/test_decompositions.py

```diff
@@ -153,6 +155,18 @@ def test_QR():
     assert R.is_upper
     assert A == Q*R
 
+    x = Symbol('x')
```

Comment:
probably should be `x = Symbol('x', nonzero=True)`?  Otherwise, the `Q` matrix is not well-defined.

Reply:
...and would the original code have worked in that case?  That suggests that a better fix may be possible.

Reply:
It does seem suspicious to be able to take a path through this GS/MGS and get empty (degenerate) matrices.  

Here's another case to worry about:
```
Matrix([[0]]).QRdecomposition()
Out[42]: (Matrix(1, 0, []), Matrix(0, 1, []))
```
(That should give `Q=[[1]]` and `R=[[0]]`.)

Reply:
Oh I see, this is a "thin" rank-revealing QR factorization.

```
>>> q, r = Matrix([[0]]).QRdecomposition()
>>>: q
Matrix(2, 0, [])
>>> r
Matrix(0, 2, [])

>>> q*r
Matrix([[0]])
```
I also tested with 2x2 zero matrix and it also works!  That is pretty neat!  At least some of this seems to be unit-tested too.

Reply:
I've attempted to use the preceding boilerplates for tests, but sympy is incapable of proving 'Q * Q.H == I' because 'x * conjugate(x) / Abs(x)**2' is not simplified even with 'simplify'